bin/jsize

```diff
@@ -7,61 +7,119 @@ const pkg = require('../package.json')
+// Store results for rendering later.
+const bundles = []
+const headings = ['.', '.']
+  .map(it => chalk.hidden(it))
+  .concat(['initial', 'minified', 'gzipped'].map(
+    heading => chalk.bold.dim(heading)
+  ))
 
 if (program.args.length > 0) {
+  // Start cli render loop.
+  const interval = setInterval(render, 100)
+
   // Build and analyze packages sequentially.
   program
     .args
-    .reduce((p, pkgs) => p.then(() => {
-      spinner.text = colorName(pkgs)
-      spinner.start()
-      return jsize(pkgs.split('+'))
-        .then(print)
-        .catch(err => spinner.fail(err.message))
-    }), Promise.resolve())
-    .catch(err => console.error(err))
+    .reduce((p, pkgs) => {
+      const bundle = {
+        name: pkgs.replace(/\+/g, chalk.dim(' + ')),
+        stats: null
+      }
+      bundles.push(bundle)
+
+      return p.then(() => {
+        return jsize(pkgs.split('+'))
+          .then(result => { bundle.stats = result })
+          .catch(err => { bundle.error = err.message })
+      })
+    }, Promise.resolve())
+    .then(() => {
+      clearInterval(interval)
+      render()
+
+      const err = bundles
+        .map(bundle => bundle.error)
+        .filter(Boolean)
+        .join('\n')
+
+      if (err) {
+        console.error(chalk.red(err))
+        process.exit(1)
+      }
+    })
+    .catch(err => {
+      console.error(err)
+      process.exit(1)
+    })
 } else {
   program.help()
 }
 
 /**
- * Reducer to calculate the max length.
- *
- * @param {number} max - the current maximum.
- * @param {string|array} it - the item being tested.
- * @return {number}
+ * Renders a table with the current results to the terminal.
  */
-function maxLength (max, it) {
-  return Math.max(max, colorName(it).length)
-}
+function render () {
+  const loading = spinner.frame()
+  const table = bundles.map(bundle => {
+    const stats = bundle.stats
+    const row = [
+      bundle.name,
+      chalk.dim('=')
+    ]
+
+    if (stats) {
+      if (program.verbose) {
+        row.push(
+          prettyBytes(stats.initial),
+          prettyBytes(stats.minified),
+          colorBytes(stats.gzipped)
+        )
+      } else {
+        row.push(colorBytes(stats.gzipped) + ' (gzipped)')
+      }
+    } else {
+      const fill = bundle.error ? chalk.red('X') : loading
+      row.push(fill)
+
+      if (program.verbose) {
+        row.push(fill, fill)
+      }
+    }
+
+    return row
+  })
 
-/**
- * Logs stats from jsize results.
- *
- * @param {object} result - the sizes from jsize.
- */
-function print (result) {
-  let info = ''
   if (program.verbose) {
-    info += padEnd(prettyBytes(result.initial) + ' (initial)', 20, ' ')
-    info += padEnd(prettyBytes(result.minified) + ' (minified)', 20, ' ')
+    table.unshift(headings, padding)
   }
-  info += colorBytes(result.gzipped) + ' (gzipped)'
-  spinner.succeed(padEnd(spinner.text, padding, ' ') + info)
+
+  // Add padding to table.
```

Comment:
What about adding padding from left? Two spaces for each row – will be looking better.

Reply:
I'll try it out.